src/main/drivers/system_stm32h7xx.c

```diff
@@ -102,7 +102,7 @@ void systemReset(void)
 
 void forcedSystemResetWithoutDisablingCaches(void)
 {
-    persistentObjectWrite(PERSISTENT_OBJECT_RESET_REASON, RESET_FORCED);
+    // Don't overwrite the PERSISTENT_OBJECT_RESET_REASON; just make another attempt
```

Comment:
It an issue that if a reset if forced for some reason (and this seems to happen quite a lot during clock initialisation) then I still want the FC to come up in the intended mode. This is saying "retry" rather than "abort" which is much more user friendly, regardless of the mode being entered, but it was impacting MSC for me.

Reply:
See USE_H7_HSE_TIMEOUT_WORKAROUND

Reply:
@mikeller Does this change make sense now?

Reply:
Ok, makes sense.